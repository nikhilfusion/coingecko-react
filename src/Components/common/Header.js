// This is the header page common to all routes
import React from 'react';
import { AppBar, Toolbar, Typography, Button, IconButton, makeStyles } from '@material-ui/core';
import { Menu as MenuIcon } from '@material-ui/icons';

const useStyles = makeStyles((theme) => ({
  root: {
    flexGrow: 1,
  },
  menuButton: {
    marginRight: theme.spacing(2),
  },
  title: {
    flexGrow: 1,
  },
}));

const Header = () => {
  const classes = useStyles();
  return (
    <div className={classes.root}>
      <AppBar position="fixed">
        <Toolbar>
          <IconButton href="/" edge="start" className={classes.menuButton} color="inherit" aria-label="menu">
            <MenuIcon />
          </IconButton>
          <Typography variant="h6" className={classes.title} data-testid="title">
            Crypto App
          </Typography>
          <Button color="inherit" data-testid="login">Login</Button>
        </Toolbar>
      </AppBar>
    </div>
  )
}

export default Header;
