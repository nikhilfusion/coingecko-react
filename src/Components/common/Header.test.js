import React from 'react';
import Header from './Header';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';

test("Header renders with correct text", () => {
  const { getByTestId } = render(<Header />);
  const headerEl = getByTestId("title");
  expect(headerEl.textContent).toBe("Crypto App");
})

test("Check Login button exist", () => {
  render(<Header />);
  expect(screen.getByRole('button')).toHaveTextContent('Login');
})

test('it renders correctly', () => {
  const tree = renderer.create(<Header />).toJSON();
  expect(tree).toMatchSnapshot()
})

