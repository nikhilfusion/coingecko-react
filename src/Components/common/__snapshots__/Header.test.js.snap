// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly 1`] = `
<div
  className="makeStyles-root-7"
>
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary mui-fixed MuiPaper-elevation4"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <a
        aria-disabled={false}
        aria-label="menu"
        className="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-8 MuiIconButton-colorInherit MuiIconButton-edgeStart"
        href="/"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
        </span>
      </a>
      <h6
        className="MuiTypography-root makeStyles-title-9 MuiTypography-h6"
        data-testid="title"
      >
        Crypto App
      </h6>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
        data-testid="login"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Login
        </span>
      </button>
    </div>
  </header>
</div>
`;
